src/realm/list.cpp

```diff
@@ -135,28 +135,38 @@ template <class T>
 void Lst<T>::sort(std::vector<size_t>& indices, bool ascending) const
 {
     update_if_needed();
-    auto tree = m_tree.get();
-    if (ascending) {
-        do_sort(indices, size(), [tree](size_t i1, size_t i2) noexcept {
-            return tree->get(i1) < tree->get(i2);
-        });
-    }
-    else {
-        do_sort(indices, size(), [tree](size_t i1, size_t i2) noexcept {
-            return tree->get(i1) > tree->get(i2);
-        });
-    }
+    auto comparator = [this, &ascending](size_t i1, size_t i2) {
+        if constexpr (std::is_same_v<T, Mixed>) {
+            return ascending ? get(i1) < get(i2) : get(i1) > get(i2);
+        }
+        else {
+            return ascending ? m_tree->get(i1) < m_tree->get(i2) : m_tree->get(i1) > m_tree->get(i2);
```

Comment:
Having the ascending check inline like this make for smaller code, but it also adds an extra check for every comparison which will be worse for performance. It is going to be somewhat ugly code to break it into four different cases, but I think we want to do that for the sake of performance.

Reply:
I see your point..